internal/ctl/connectors/add.go

```diff
@@ -44,16 +46,25 @@ func doAddConnectors(_ *cobra.Command, params *addConnectorsCmdParams) {
 	clusterLogger.WithField("config", config).Trace("add connectors configuration")
 
 	var source manager.ConnectorSource
-	if params.Files != nil {
-		source = filesSource(&params.Files)
-	}
-	if params.Directory != "" {
-		source = directorySource(&params.Directory)
+
+	switch {
+	case params.Files != nil:
+		if len(params.Files) == 1 && params.Files[0] == "-" {
+			source = sources.StdIn(cmd.InOrStdin())
+		} else {
+			source = sources.Files(params.Files)
+		}
+	case params.Directory != "":
+		source = sources.Directory(params.Directory)
+	case params.EnvVar != "":
+		source = sources.EnvVarValue(params.EnvVar)
+	default:
+		clusterLogger.Fatalln("error finding connector definitions from parameters")
```

Comment:
Could we avoid using Fatal ...?

Reply:
I think I know what to do to make this happen - I'll do it on another PR